src/components/progress-bar/components/RailsList.tsx

```diff
@@ -0,0 +1,64 @@
+import { FC, memo } from 'react';
+import { uuid } from 'uuidv4';
+
+import { VideoContext } from '../../../context';
+import { Highlight } from '../../../types';
+import {
+	getPercentFromDuration,
+	getRailSegments,
+} from '../../../utils/highlights';
+
+import { RailStyled } from './RailStyled';
+
+interface RailListProps {
+	highlights: Highlight[];
+	videoDuration: number;
+	getHighlightColorBlended?: VideoContext['getHighlightColorBlended'];
+}
+
+export const createSegmentColor = (
+	colors?: string[],
+	getHighlightColorBlended?: VideoContext['getHighlightColorBlended'],
+) => {
+	if (colors && colors?.length > 0 && getHighlightColorBlended) {
+		return getHighlightColorBlended(colors);
+	}
+	return undefined;
+};
+
+export const RailsList: FC<RailListProps> = memo(
+	({ highlights, videoDuration, getHighlightColorBlended }) => {
+		const railSegments = getRailSegments(highlights, videoDuration);
```

Comment:
This business logic looks quite complex. Could it be moved into a hook and add tests for it?

Reply:
added a util function that creates rails props from segments